app/services/metrc_service/plant/move.rb

```diff
@@ -18,36 +17,25 @@ def transaction
       end
 
       def prior_move
-        return @prior_move if @prior_move
-
-        previous_move = Transaction.where(
-          'batch_id = ? AND type = ? AND vendor = ? AND id NOT IN (?)',
-          @batch_id,
-          :move_batch,
-          :metrc,
-          transaction.id
-        ).limit(1).order('created_at desc').first
+        previous_move = batch.completions.select do |comp|
+          comp.action_type == 'move' && comp.id < @completion_id
+        end.max_by(&:start_time)
 
         return if previous_move.nil?
 
-        @prior_move = get_completion(previous_move&.completion_id)
+        # calling get_completion here will ensure relationships are side loaded.
+        get_completion(previous_move.id)
       end
       memoize :prior_move
 
       def prior_start
-        return @prior_start if @prior_start
-
-        previous_start = Transaction.where(
-          'batch_id = ? AND type IN (?) AND vendor = ? AND id NOT IN (?)',
-          @batch_id,
-          %i[start_batch start_batch_from_package start_batch_from_source_plant],
-          :metrc,
-          transaction.id
-        ).limit(1).order('created_at desc').first
-
+        previous_start = batch.completions.select do |comp|
```

Comment:
a batch should not have more than one start... so this whole `prior_start` method should be renamed to `start_completion` and simplified

Reply:
👍